src/main/java/net/jodah/failsafe/internal/util/CircularBitSet.java

```diff
@@ -22,8 +22,24 @@ public CircularBitSet(int size, CircularBitSet oldBitSet) {
 
     // Initialize from oldBitSet
     if (oldBitSet != null) {
-      for (int i = 0; i < size && i < oldBitSet.occupiedBits; i++)
+      initializeFromOldBits(oldBitSet);
+    }
+  }
+
+  /**
+   * Initialize bitSet from oldBitSet with correct order
+   */
+  private void initializeFromOldBits(CircularBitSet oldBitSet) {
```

Comment:
It's very nice! Thanks for the advice. I had just update it.

But I just think another situation.
There is order issue in situation that new BitSet had larger size or same size than old BitSet.

Here is the unit test!
```java
  public void shouldCopyBitsToLargerBitSetWithCorrectOrder() {
    CircularBitSet bs = new CircularBitSet(2, null);
    boolean[] prevBits = { true, true, false };
    for (boolean prevBit : prevBits) {
      bs.setNext(prevBit);
    }

    CircularBitSet bs2 = new CircularBitSet(2, bs); // expected { true, false }
    assertEquals(bs2.occupiedBits(), 2);
    assertEquals(bs2.positiveRatio(), .5);
    assertEquals(bs2.negativeRatio(), .5);

    bs2.setNext(true); // expected { false, true }
    assertEquals(bs2.occupiedBits(), 2);
    assertEquals(bs2.positiveRatio(), .5);
    assertEquals(bs2.negativeRatio(), .5);

  }
```